e2e/src/editCra.e2e-spec.ts

```diff
@@ -0,0 +1,187 @@
+  it('should be possible to enter the client\'s name', () => {
+    clientInput.sendKeys('Test.com');
+    expect(clientInput.getAttribute('value')).toBe('Test.com');
+  });
+
+  describe('On the calendar', () => {
+    let aDayCell: ElementFinder;
+
+    beforeEach(() => {
+      aDayCell = editCraPage.getElementsByText('4').first();
+      aDayCell.click();
+    });
+
+    it('should be possible to select days', () => {
+      expect(editCraPage.getADayBadge(aDayCell).getText()).toBe('1\njour');
+    });
+
+    it('should be possible to select every working day', () => {
+      editCraPage.getElementsByText('Sélectionner les jours ouvrés').first().click();
+      expect(selectedNumber.getText()).toBe(getBusinessDaysNumber().toString());
+    });
+
+    describe('', () => {
+      let dayBadge: ElementFinder;
+
+      beforeEach(() => {
+        dayBadge = editCraPage.getADayBadge(aDayCell);
+
+        browser.actions()
+            .mouseMove(dayBadge)
+            .perform();
+
+        // Sleeping to wait for the annimation end
+        browser.sleep(50);
+
+        browser.actions()
+            .click(editCraPage.getElementsByText('0,5').first())
+            .perform();
+
+        // Sleeping to wait for the annimation end
+        browser.sleep(50);
+
+        // Moving the mouse away to read the value
+        browser.actions()
+            .mouseMove(aDayCell)
+            .perform();
+
+        // Sleeping to wait for the annimation end
+        browser.sleep(170);
+      });
+
+      it('should be possible to change the duration of a day', () => {
+        expect(dayBadge.getText()).toBe('0,5\njour');
+      });
+
+      it('should show the total number of days', () => {
+        editCraPage.getElementsByText('10').first().click();
+        expect(selectedNumber.getText()).toBe('1,5');
+      });
+    });
+  });
+
+  describe('When I validate my inputs', () => {
+    let submitButton: ElementFinder;
+
+    beforeEach(() => {
+      submitButton = editCraPage.getSubmitButton();
+    });
+
+    describe('If something is wrong', () => {
+      beforeEach(() => {
+        submitButton.click();
+      });
+
+      it('should tell me via a modal', () => {
+        expect(editCraPage.getElementsByText('Validation impossible').count()).toBe(1);
+      });
+
+      it('should tell me my mistakes via validation messages', () => {
+        expect(editCraPage.getElementsByText('Ce champ est obligatoire').count()).toBe(4);
+      });
+    });
+
+    describe('If everything is good', () => {
+      beforeEach(() => {
+        nameInput.sendKeys('Tester');
+        emailInput.sendKeys('tester@test.com');
+        titleInput.sendKeys('Testing');
+        clientInput.sendKeys('Test.com');
+        editCraPage.getElementsByText('Sélectionner les jours ouvrés').first().click();
+        submitButton.click();
+      });
+
+      it('should open a modal', () => {
+        expect(editCraPage.getElementsByText('Votre CRA a bien été créé').count()).toBe(1);
```

Comment:
Better to hook on an ID, class or something else, here if you change your text, you add english trad or something else your test will fail :(